File: backend/app/services/transport_service.py
"""Faithful Python port of src/services/transportService.ts.

Estimation-only, exactly as the frontend comment states: no real transport /
booking API is connected, every figure is clearly an estimate, and no
vehicle is ever claimed to be booked. isEstimate is always True.
"""

from app.mock_data.transport import PER_PERSON_INDIVIDUAL_COST, SHARED_SAVINGS_RATIO, vehicle_for


def estimate_transport(group_size: int) -> dict:
    individual_total = group_size * PER_PERSON_INDIVIDUAL_COST
    shared_total = round(individual_total * SHARED_SAVINGS_RATIO)
    vehicle_info = vehicle_for(group_size)

    savings_total = individual_total - shared_total
    savings_per_person = round(savings_total / group_size) if group_size else 0

    return {
        "group_size": group_size,
        "individual": {"ride_count": group_size, "total_cost": individual_total},
        "shared": {
            "vehicle": vehicle_info["vehicle"],
            "capacity": vehicle_info["capacity"],
            "total_cost": shared_total,
            "cost_per_person": round(shared_total / group_size) if group_size else 0,
        },
        "savings_total": savings_total,
        "savings_per_person": savings_per_person,
        "is_estimate": True,
    }

File: frontend/src/services/transportService.ts
import type { TransportComparison } from '@/types/transport'
import { simulateLatency } from './api'

// Estimation-only. No real transport / booking API is connected in this
// prototype, so every figure here is clearly labelled "Estimated" in the UI
// and no vehicle is ever claimed to be booked.

const PER_PERSON_INDIVIDUAL_COST = 1000 // average local transport spend/person across the trip
const SHARED_SAVINGS_RATIO = 0.6 // shared travel costs ~60% of the individual total

const vehicleFor = (groupSize: number): { vehicle: string; capacity: number } => {
  if (groupSize <= 4) return { vehicle: 'Shared Cab', capacity: 4 }
  if (groupSize <= 8) return { vehicle: 'Mini-Bus', capacity: 8 }
  return { vehicle: 'Tempo Traveller', capacity: 12 }
}

export async function estimateTransport(groupSize: number): Promise<TransportComparison> {
  await simulateLatency(400)

  const individualTotal = groupSize * PER_PERSON_INDIVIDUAL_COST
  const sharedTotal = Math.round(individualTotal * SHARED_SAVINGS_RATIO)
  const { vehicle, capacity } = vehicleFor(groupSize)

  const savingsTotal = individualTotal - sharedTotal
  const savingsPerPerson = Math.round(savingsTotal / groupSize)

  return {
    groupSize,
    individual: { rideCount: groupSize, totalCost: individualTotal },
    shared: {
      vehicle,
      capacity,
      totalCost: sharedTotal,
      costPerPerson: Math.round(sharedTotal / groupSize),
    },
    savingsTotal,
    savingsPerPerson,
    isEstimate: true,
  }
}
